src/sources/csv.rs

```diff
@@ -0,0 +1,261 @@
+//! CSV source for syncing with Famedly's Zitadel.
+
+use std::{fs, path::PathBuf};
+
+use anyhow::{Context, Result};
+use async_trait::async_trait;
+use csv::Reader;
+use serde::Deserialize;
+
+use super::Source;
+use crate::{user::User, zitadel::SourceDiff};
+
+/// CSV Source
+pub struct CsvSource {
+	/// CSV Source configuration
+	csv_config: CsvSourceConfig,
+}
+
+#[async_trait]
+impl Source for CsvSource {
+	fn get_name(&self) -> &'static str {
+		"CSV"
+	}
+
+	async fn get_diff(&self) -> Result<SourceDiff> {
+		let new_users = self.read_csv()?;
+		return Ok(SourceDiff { new_users, changed_users: vec![], deleted_user_ids: vec![] });
+	}
+}
+
+impl CsvSource {
+	/// Create a new CSV source
+	pub fn new(csv_config: CsvSourceConfig) -> Self {
+		Self { csv_config }
+	}
+
+	/// Get list of users from CSV file
+	fn read_csv(&self) -> Result<Vec<User>> {
+		let file_path = &self.csv_config.file_path;
+		let file = fs::File::open(&self.csv_config.file_path)
+			.context(format!("Failed to open CSV file {}", file_path.to_string_lossy()))?;
+		let mut reader = Reader::from_reader(file);
+		let users: Result<Vec<CsvData>, _> = reader.deserialize().collect();
+		let users: Result<Vec<User>, _> = users
+			.context("Failed to deserialize CSV data")
+			.map(|users| users.into_iter().map(CsvData::to_user).collect());
+		users
+	}
+}
+
+/// Configuration to get a list of users from a CSV file
+#[derive(Debug, Clone, Deserialize, PartialEq)]
+pub struct CsvSourceConfig {
+	/// The path to the CSV file
+	pub file_path: PathBuf,
+}
+
+/// CSV data structure
+#[derive(Debug, Deserialize)]
+struct CsvData {
+	/// The user's email address
+	email: String,
+	/// The user's first name
+	first_name: String,
+	/// The user's last name
+	last_name: String,
+	/// The user's preferred username
+	preferred_username: String,
+	/// The user's phone number
+	phone: String,
+	/// Whether the user is enabled
+	enabled: bool,
+}
+
+impl CsvData {
+	/// Convert CsvData to User data
+	fn to_user(csv_data: CsvData) -> User {
+		User {
+			first_name: csv_data.first_name.into(),
+			last_name: csv_data.last_name.into(),
+			email: csv_data.email.into(),
+			phone: if csv_data.phone.is_empty() { None } else { Some(csv_data.phone.into()) },
+			enabled: csv_data.enabled,
+			preferred_username: csv_data.preferred_username.clone().into(),
+			ldap_id: csv_data.preferred_username.into(),
+		}
+	}
+}
+
+/// Helper module for unit and e2e tests
+pub mod test_helpers {
```

Comment:
`#[cfg(test)]` maybe?

Reply:
We should add `#![cfg(test)]` to those (as in https://github.com/famedly/backend-system-tests/blob/ef787dc3d6a882e18f1333abf9c5ee73e7935151/tests/zitadel.rs#L1)

Reply:
rustc doesn't like that and says `#![no_std]` above the test_helpers module and says inner attributes are usually found at the beginning of source files

Reply:
I mean we can put `#![cfg(test)]` to the top of the `test/*.rs` modules, then they should compile if `test_helpers` is feature gated by `#[cfg(test)]`

Reply:
The thing is that those modules have to be re-exported from lib.rs, which gives: found an item that was configured out. Am I missing something?

Reply:
you're right, that's impossible